worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocument.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.worker.markup.core;
+
+import com.github.cafdataprocessing.worker.markup.core.Hashing.HashHelper;
+import com.github.cafdataprocessing.worker.markup.core.exceptions.MarkupWorkerExceptions;
+import com.google.common.collect.Multimap;
+import com.hpe.caf.api.Codec;
+import com.hpe.caf.api.ConfigurationException;
+import com.hpe.caf.api.ConfigurationSource;
+import com.hpe.caf.api.worker.DataStore;
+import com.hpe.caf.api.worker.DataStoreSource;
+import com.hpe.caf.codec.JsonCodec;
+import com.hpe.caf.util.ref.DataSource;
+import com.hpe.caf.util.ref.ReferencedData;
+import com.hpe.caf.worker.document.model.Document;
+import com.hpe.caf.worker.markup.HashConfiguration;
+import com.hpe.caf.worker.markup.MarkupWorkerResult;
+import com.hpe.caf.worker.markup.MarkupWorkerStatus;
+import com.hpe.caf.worker.markup.MarkupWorkerTask;
+import com.hpe.caf.worker.markup.OutputField;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.jdom2.JDOMException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MarkupDocument
+{
+    private static final Logger LOG = LoggerFactory.getLogger(MarkupDocument.class);
+
+    final ExecutorService jepThreadPool = Executors.newSingleThreadExecutor();
```

Comment:
I see a change in behaviour wrt the previous implementation. Thread pool in here will be created for each request while previously it would only be created once per worker. Thread pool is also not shut down.
I think we should have a constructor which can take in ExecutorService and control it in the worker factory as previously. How is the executor service used? What are implications of re-creating it every time on the request? These are the questions to ask before making a change like that...